src/perturbopy/postproc/utils/spectra_plots.py

```diff
@@ -0,0 +1,128 @@
+        ax.scatter(hole_kpoint_array[idx, 2], hole_energy_array[idx, i], s=h_occs[idx, i][0] * 1000 + 1e-10, c='red',
+                   alpha=0.5)
+
+    fsize = 16
+    ax.set_title(f'Occupation amplitude for pump energy {pump_energy:.3f} eV')
+    ax.set_xlabel('Electron Momentum $k_x$')
+    ax.set_ylabel('Energy (eV)')
+
+    plt.show()
+
+
+def animate_pump_pulse(time_step,
+                       elec_delta_occs_array, elec_kpoint_array, elec_energy_array,
+                       hole_delta_occs_array, hole_kpoint_array, hole_energy_array,
+                       pump_energy):
+    """
+    Animate the pump pulse excitation for electrons and holes.
+    """
+
+    fig, ax = plt.subplots(1, 1, figsize=(9, 6))
+
+    idx_elec = np.where((elec_kpoint_array[:, 1] == 0) & (elec_kpoint_array[:, 0] == 0))
+    idx_hole = np.where((hole_kpoint_array[:, 1] == 0) & (hole_kpoint_array[:, 0] == 0))
+
+    # Plot electron occupations
+    elec_scat_list = []
+    for i in range(np.size(elec_energy_array, axis=1)):
+        ax.scatter(elec_kpoint_array[idx_elec, 2], elec_energy_array[idx_elec, i], s=0.5, c='black', alpha=0.5)
+        scat_obj = ax.scatter(elec_kpoint_array[idx_elec, 2], elec_energy_array[idx_elec, i], s=0.0, c='red', alpha=0.5)
+        elec_scat_list.append(scat_obj)
+
+    # Plot hole occupations
+    hole_scat_list = []
+    for i in range(np.size(hole_energy_array, axis=1)):
+        ax.scatter(hole_kpoint_array[idx_hole, 2], hole_energy_array[idx_hole, i], s=0.5, c='black', alpha=0.5)
+        obj = ax.scatter(hole_kpoint_array[idx_hole, 2], hole_energy_array[idx_hole, i], s=0, c='red', alpha=0.5)
+        hole_scat_list.append(obj)
+
+    fsize = 16
+    ax.set_title(f'Pump energy {pump_energy:.3f} eV; Time: 0.0 fs')
+    ax.set_xlabel('Electron Momentum $k_x$')
+    ax.set_ylabel('Energy (eV)')
+
+    ani = FuncAnimation(fig, update_scatter, frames=elec_delta_occs_array.shape[1],
+                        fargs=(ax, time_step, idx_elec, idx_hole,
+                               elec_scat_list, hole_scat_list,
+                               elec_delta_occs_array, elec_kpoint_array, elec_energy_array,
+                               hole_delta_occs_array, hole_kpoint_array, hole_energy_array),
+                        interval=100, repeat=False)
+
+    # Save the animation to gif
+    ani.save('pump_pulse.gif', writer='imagemagick')
+
+    plt.show()
+
+
+def update_scatter(anim_time, ax, time_step, idx_elec, idx_hole,
+                   elec_scat_list, hole_scat_list,
+                   elec_delta_occs_array, elec_kpoint_array, elec_energy_array,
+                   hole_delta_occs_array, hole_kpoint_array, hole_energy_array):
+    """
+    Animate the pump pulse excitation for electrons and holes.
+
+    Parameters
+    ----------
+
+    anim_time : float
+        Animation time.
+
+    ax : matplotlib.axes.Axes
+        Axis for plotting the pump pulse excitation.
+    """
```

Comment:
Hi Ivan, everything looks great! I think my only comment is adding more documentation about the input parameters for these functions.

Reply:
I did it, thank you Laurie.

We should however discuss and establish some policies for documentation of functions in Perturbopy. 
Cases we sould try to avoid is for one-line function having 20 lines of documentation. This could make the code actuallly less readable.

The main functions and modules must be very well documented. Other functions, less. I still documented entirely, to make the PR asap and to postpone this question after the code release :)
Also, I am very well aware that if we ease this rule now, some people can stop documenting everything, so we have to intoduce this rule with caution :) 

 @hurricane642 @alex2shiyu 
 
 Below is a screenshot of a function, where the comments, in my opinion, make it less readable
 
<img width="815" alt="image" src="https://github.com/user-attachments/assets/a60becb6-9ac9-4a57-b349-c74317f21bf7" />

Reply:
That's interesting question for discussion. Maybe be we can try to make a convention for documentation of this type of internal functions in the more condensed and short matter. We'll think about this later, but @imaliyov , thank you a lot for documenting all of this now, that is great!